src/types.rs

```diff
@@ -519,6 +594,20 @@ impl dyn Interface {
         )?)?)
     }
 
+    pub fn get_data_wasmv1<T: DeserializeOwned>(
+        &self,
+        key: &[u8],
+        address: Option<NativeAddress>,
+    ) -> Result<T> {
+        Ok(serde_json::from_str::<T>(std::str::from_utf8(
+            &self.raw_get_data_wasmv1(key, address)?,
+        )?)?)
+    }
+
+    pub fn set_data<T: Serialize>(&self, key: &[u8], value: &T) -> Result<()> {
```

Comment:
same as above: protobuf is probably enough

Reply:
Added a TODO comment